.github/ISSUE_TEMPLATE/bug.yaml

```diff
@@ -0,0 +1,60 @@
+
+name: 🐞 Bug
+description: Report an issue to help improve the project.
+title: "[BUG] <title>"
+labels: ["bug"]
+body:
+  - type: checkboxes
+    attributes:
+      label: Is there an existing issue for this?
+      description: Please search to see if an issue already exists for the bug you encountered.
+      options:
+      - label: I have searched the existing issues
+        required: true
+  - type: textarea
+    id: description
+    attributes:
+      label: Description
+      description: A brief description of the question or issue, also include what you tried and what didn't work
+    validations:
+      required: true
+  - type: textarea
+    id: Reproduce
+    attributes:
+      label: To Reproduce
+      description: Steps to reproduce this bug
+    validations:
+      required: true
+  - type: textarea
+    id: screenshots
+    attributes:
+      label: Screenshots
+      description: Please add screenshots if applicable
+    validations:
+      required: false
+  - type: textarea
+    id: extrainfo
+    attributes:
+      label: Additional information
+      description: Is there anything else we should know about this bug? (like linux distribution(version + arch), python version and etc)   
+    validations:
+      required: false
+  - type: dropdown
+    id: component
+    attributes:
+      label: Where did you see this issue?
+      description: In which pbench component this issue was found?
+      options:
+        - Agent
+        - Server
+        - Other
```

Comment:
I did some research on having this automated, like when user would choose one of this dropdown options then that would automatically add respective labels as well. But yaml parser of GitHub issue forms is not intelligent enough to do that (i tried declaring variable as we do in ansible playbook and reuse but no success).

I don't feel that would really be a burden on user, infact we are trying to easy bug reporting by making it more interactive and informative (for end-user and team member).

Reply:
I agree that a pull-down is pretty easy to use.  The problem is that I don't think it produces much value here.  In the face of that, it takes up space on the screen (whether in the YML file or the rendered form), and it requires a response from the user, in order to produce a result which the team members probably won't pay much attention to.  Whoever ends up triaging the Issue will read the description and apply the appropriate label, so forcing the author to decide which option to pick (especially if s/he isn't sure) is a waste of time and effort.

Reply:
Agreed team members would take no time to identify which component this issue belongs to.
But I think this is an opensource project and we will have many public contribution in future where people can contribute in triaging tickets, bug fixes and much more.
So in that case project should not be dependent on only team members for triaging and others activities.
Also if any new contributor need to pick any issue for bug fix which is not triaged it would be difficult for him/her to have a idea where to look for.

And bug reports should be clear and precise with all necessary information included. There shouldn't be anything in the bug that should be assumed by the project maintainers.

Reply:
I like forcing the user to choose from a menu where it makes any sense, because it

1. forces a response, and
2. limits the response to a simple and sensible set of values

I thought this was a good application since we have a few major components; however on further consideration the big problem is that we could easily have issues that cross boundaries (server and/or dashboard, agent and/or server, etc.) and I don't think these support multiple selection.

So, yeah; an empty field that's required is probably better.

Reply:
@dbutenhof  we can have multi selection dropdown, just need to add `multiple: true` option

Reply:
> @dbutenhof we can have multi selection dropdown, just need to add `multiple: true` option

OK, that's interesting. While Webb's correct that whoever does the triage will need to analyze the defect report to determine the actual project classification/labels anyway, I like the idea of this selector to focus the user's description.

Maybe change the description to something more like "I encountered this problem while interacting with the ..." [agent, server, other]  We could in theory expand this to include fio, uperf, etc., but maybe that's too much. I also like the idea of including "dashboard," and once more it bugs me that we're managing that out of a separate GitHub repo.

In any case, in the longer term we're moving towards *planning* out of Jira rather than GitHub issues, so these will serve as an external entry point to our defect pipeline; the first step in triage will probably be for a team member to translate the defect into a Jira ticket. (And, yes, we're going to have to figure out how this is all going to work if we really do manage to develop a community outside of Red Hat; but planning out of GitHub is just too chaotic.)

Reply:
I guess I'm just having trouble understanding what problem having this pulldown is supposed to solve.  The author of the Issue is going to include a description the problem...if the reader cannot tell from that which component the problem is in, then we've already got a challenge which this pulldown is unlikely to resolve.  When it comes time for contributors to pick up tasks, I expect that they will filter the Issues list based on GitHub labels, not based on searching for values in this field.  So, given that this field doesn't result in labels (except via triage, which I expect will be based on the description and not so much this field), I don't see how this field produces any value.

And, my foremost concern (possibly unfounded...) is that the author might not know or otherwise be able to correctly select which component(s) correspond to the problem...in which case, I don't want the contents of this field to become a misdirection.  That is, determining the value for this field is a triage or analysis action, and, while it's great if the author is capable doing this, it's not their role, and, if they get it wrong (even if they are well-meaning), then the result is counter-productive.

So, I think we're better off just omitting this from the form.  (Or, at the very least, setting its value should be optional, so that the author doesn't **have** to make a _guess_.)

Reply:
My perspective is that I see too many defect reports that assume virtually everything about the tester's environment and don't give enough information. Sure, sometimes we can guess, and at worst we say "sorry, can't work on this without more information", but I'd rather provide as many prompts for actionable information as possible up front to encourage good reporting. I think that often being forced to think about identifying what they were doing in a pulldown list helps to focus their thinking, if nothing else.

And if they really can't tell, there's always "other" ... we could add a "I don't know" although that might be too much of a temptation. 😬 

Whether this specific prompt, or these specific choices, will turn out to be valuable is a different matter, but I'd rather try something than assume it won't work.

Reply:
My motivation is to make sure that we actually receive a report when there is a problem.  I don't want the would-be report author to look at the form and say to him-/herself, "Whoah, this isn't worth it!" and then be tempted to walk away.  (I know I've had that experience....)  So, I want the form to be as simple and easy as possible while still standing a good chance of capturing what we need.  If the report is insufficient, at least we are aware that there was a problem and we stand a chance of being able to get additional information, which won't be the case if the reporter didn't bother to make the report.

Now, granted, a single pull-down is unlikely to make or break anything, but, in an attempt to avoid the death of a thousand cuts, this item seems like one that we could jettison, especially since it seems unlikely (to me...) to produce anything beyond what's already in the description.

I'm sympathetic to the desire to avoid reports with insufficient details (but whether the _author_ thinks that the problem is in the Agent or the Server doesn't strike me as one that we're likely to care about).  I concur that it likely would be helpful to prompt the author to provide us with specific information, but I would do that via a placeholder or instructional text in the form, rather than having specific, required fields for each (with or without pulldowns).

I'm inclined to try this without the pulldown, under the idea that the "big three" ("What did you do?" "What happened?" "What did you expect to happen?") will be sufficient...and, if they aren't then we can iterate.